server/src/main/java/org/elasticsearch/search/query/EarlyTerminatingCollector.java

```diff
@@ -59,29 +59,28 @@ public ScoreMode scoreMode() {
     @Override
     public LeafCollector getLeafCollector(LeafReaderContext context) throws IOException {
         if (numCollected >= maxCountHits) {
-            earlyTerminated = true;
-            if (forceTermination) {
-                throw new EarlyTerminationException("early termination [CountBased]");
-            } else {
-                throw new CollectionTerminatedException();
-            }
+            earlyTerminate();
         }
         return new FilterLeafCollector(super.getLeafCollector(context)) {
             @Override
             public void collect(int doc) throws IOException {
                 if (++numCollected > maxCountHits) {
-                    earlyTerminated = true;
-                    if (forceTermination) {
-                        throw new EarlyTerminationException("early termination [CountBased]");
-                    } else {
-                        throw new CollectionTerminatedException();
-                    }
+                    earlyTerminate();
                 }
                 super.collect(doc);
             }
         };
     }
 
+    private void earlyTerminate() {
+        earlyTerminated = true;
+        if (forceTermination) {
+            throw new EarlyTerminationException("early termination [CountBased]");
+        } else {
+            throw new CollectionTerminatedException();
+        }
+    }
```

Comment:
minor adjustment, I found it easier to parse knowing that the two code blocks do exactly the same.